fuel-core-interfaces/src/relayer.rs

```diff
@@ -0,0 +1,143 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use fuel_storage::Storage;
+use fuel_types::{Address, AssetId, Bytes32, Word};
+use serde::{Deserialize, Serialize};
+use tokio::sync::oneshot;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DepositCoin {
+    pub owner: Address,
+    pub amount: Word,
+    pub asset_id: AssetId,
+    pub eth_block_deposited: u64,
+    pub fuel_block_spend: u64,
```

Comment:
```suggestion
    pub fuel_block_spend: Option<u64>,
```

Reply:
I assuming that `fuel_block_spend` is zero if it is not spend, and with that we are "saving" some space. I  am okay with leaving it as it is now or adding Option if you like it more.

Reply:
True, the option would add an extra byte. However, I think deposits could be pruned eventually so maybe we can afford to be less conservative and have the types be more expressive.

Reply:
Changed it to Option